src/wasapi.c

```diff
@@ -5,11 +5,94 @@
  * See http://opensource.org/licenses/MIT
  */
 
+#define INITGUID
+#define CINTERFACE
+#define COBJMACROS
+#define CONST_VTABLE
+#include <initguid.h>
+#include <audioclient.h>
+#include <endpointvolume.h>
+#include <mmdeviceapi.h>
+#include <mmreg.h>
+#include <functiondiscoverykeys_devpkey.h>
+
 #include "wasapi.h"
 #include "soundio_private.h"
 
 #include <stdio.h>
 
+#ifdef __cplusplus
+// In C++ mode, IsEqualGUID() takes its arguments by reference
+#define IS_EQUAL_GUID(a, b) IsEqualGUID(*(a), *(b))
+#define IS_EQUAL_IID(a, b) IsEqualIID((a), *(b))
+
+// And some constants are passed by reference
+#define IID_IAUDIOCLIENT                      (IID_IAudioClient)
+#define IID_IMMENDPOINT                       (IID_IMMEndpoint)
+#define IID_IAUDIOCLOCKADJUSTMENT             (IID_IAudioClockAdjustment)
+#define IID_IAUDIOSESSIONCONTROL              (IID_IAudioSessionControl)
+#define IID_IAUDIORENDERCLIENT                (IID_IAudioRenderClient)
+#define IID_IMMDEVICEENUMERATOR               (IID_IMMDeviceEnumerator)
+#define IID_IAUDIOCAPTURECLIENT               (IID_IAudioCaptureClient)
+#define CLSID_MMDEVICEENUMERATOR              (CLSID_MMDeviceEnumerator)
+#define PKEY_DEVICE_FRIENDLYNAME              (PKEY_Device_FriendlyName)
+#define PKEY_AUDIOENGINE_DEVICEFORMAT         (PKEY_AudioEngine_DeviceFormat)
+
+// And some GUID are never implemented (Ignoring the INITGUID define)
```

Comment:
For the GUIDs that we expect to find in the headers, but for some reason are missing, I like the solution of just putting the value into libsoundio. However, let's make the names in the libsoundio namespace in case MS decides sometime to fix the bug and provide the names. For example, `SOUNDIO_IID_IMMDeviceEnumerator` instead of `IID_IMMDEviceEnumerator`